src/caf/space/weighted_funcs.py

```diff
@@ -0,0 +1,181 @@
+    warnings.warn(
+        f"{missing} zones do not match up between the lower zoning and weighting data."
+    )
+    weighted["lower_area"] = weighted.area
+    return weighted
+
+
+def _create_tiles(zone_1: si.ZoneSystemInfo, zone_2: si.ZoneSystemInfo,
+                  lower_zoning: si.LowerZoneSystemInfo) -> pd.DataFrame:
+    """
+    Create a spanning set of tiles for the weighted translation.
+
+    Parameters
+    ----------
+    zone_1: Info on first zone system
+    zone_2: Info on second zone system
+    lower_zoning: Info on lower zoning and weight data
+
+    Returns
+    -------
+    A set of weighted tiles used for weighted translation.
+    """
+    zone_1 = gpd.read_file(zone_1.shapefile)
+    zone_2 = gpd.read_file(zone_2.shapefile)
+
+    LOG.info(
+        "Count of %s zone: %s",
+        zone_2.name,
+        zone_2.iloc[:, 0].count(),
+    )
+    LOG.info(
+        "Count of %s zones: %s",
+        zone_1.name,
+        zone_1.iloc[:, 0].count(),
+    )
+
+    weighting = _weighted_lower(lower_zoning)
+    tiles = reduce(
+        lambda x, y: gpd.overlay(x, y, keep_geom_type=True),
+        [zone_1, zone_2, weighting],
+    )
+    tiles.overlay_area = tiles.area
+    tiles.prop = tiles.overlay_area / tiles.lower_area
+    tiles[lower_zoning.data_col] *= tiles.prop
+    return tiles[
+        [
+            zone_1.id_col,
+            zone_2.id_col,
+            lower_zoning.data_col,
+        ]
+    ]
+
+
+def return_totals(frame: pd.DataFrame, id_col: str, data_col: str) -> pd.DataFrame:
+    """
+    Group df by dataframe and sums, keeping data_col.
+
+    Parameters
+    ----------
+    frame: A dataframe.
+    id_col: Column to group by.
+    data_col: Column to keep
+
+    Returns
+    -------
+    Grouped and summed df
+    """
+    totals = frame.groupby(id_col).sum().loc[:, data_col]
+    return totals
+
+
+def overlaps_and_totals(zone_1: si.ZoneSystemInfo, zone_2: si.ZoneSystemInfo,
+                        lower_zoning: si.LowerZoneSystemInfo) -> pd.DataFrame:
+    """
+    Create overlap totals for zone systems.
+
+    Creates totals and joins back up.
+
+    Parameters
+    ----------
+    zone_1: Info on first zone system
+    zone_2: Info on second zone system
+    lower_zoning: Info on lower zoning and weight data
+
+    Returns
+    -------
+    Dataframe with columns for overlap total, zone 1 total, zone 2 total
+    weights.
+    """
+    tiles = _create_tiles(zone_1, zone_2, lower_zoning)
+    totals_1 = return_totals(tiles,
+        zone_1.id_col, lower_zoning.data_col
+    ).to_frame()
+    totals_2 = return_totals(
+        tiles, zone_2.id_col, lower_zoning.data_col
+    ).to_frame()
+    overlap = (
+        tiles.groupby([zone_1.id_col, zone_2.id_col])
+        .sum()
+        .loc[:, lower_zoning.data_col]
+        .to_frame()
+    )
+    return overlap.join(totals_1, rsuffix="_1").join(
+        totals_2, lsuffix="_overlap", rsuffix="_2"
+    )
```

Comment:
It is explained in the function docstrings which are called but I can add comments for readability.

Reply:
Thank you! I think part of the problem comes from 2 things.

Readers are lazy - they don't want to look at the docstrings for all your function calls and `return_totals`, `totals_1`, and `tiles` are such a vague names it's hard to intuit what is happening.

Line 143 is made up of about 4 function calls in one, which can be a bit intimidating on first read. A comment explaining *why* you need to do all this would probably help massively for this line.